src/vcf/meta_entry.cpp

```diff
@@ -182,8 +182,42 @@ namespace ebi
             type_field != "String") {
             throw new MetaSectionError{entry.line, "FORMAT metadata Type is not a Integer, Float, Character or String"};
         }
+
+        if (entry.source->version == Version::v41 || entry.source->version == Version::v42) {
+            check_format_predefined_type(value, format_type_v41_v42);
+            check_format_predefined_number(value, format_number_v41_v42);
+        } else {
+            check_format_predefined_type(value, format_type_v43);
+            check_format_predefined_number(value, format_number_v43);
+        }
     }
-    
+
+    void MetaEntryVisitor::check_format_predefined_type(std::map<std::string, std::string> & value,
+                                                        std::map<std::string, std::pair<std::string, std::string>> const & format_types) const
+    {
+        auto & type_field = value["Type"];
+        for (const auto & type : format_types) {
+            if (value["ID"] == type.first) {
+                if (type_field != type.second.first) {
+                    throw new MetaSectionError{entry.line, type.second.second};
+                }
+            }
+        }
+    }
+
+    void MetaEntryVisitor::check_format_predefined_number(std::map<std::string, std::string> & value,
+                                                          std::map<std::string, std::pair<std::string, std::string>> const & format_numbers) const
+    {
+        auto & number_field = value["Number"];
+        for (const auto & number : format_numbers) {
+            if (value["ID"] == number.first) {
```

Comment:
By turning `value["Number"]` and `value["ID"]` into parameters (instead of passing a whole map) the code for `check_predefined_type` and `check_predefined_number` would be the same, and we could have a single method instead of 2.